node-graph/gstd/src/quantization.rs

```diff
@@ -0,0 +1,93 @@
+use graphene_core::raster::{Color, Image};
+use graphene_core::Node;
+
+pub struct GenerateQuantizationNode<N: Node<(), Output = u32>, M: Node<(), Output = u32>>(N, M);
+
+impl<N: Node<(), Output = u32>, M: Node<(), Output = u32>> Node<Image> for GenerateQuantizationNode<N, M> {
+	type Output = Image;
+
+	fn eval(self, input: Image) -> Self::Output {
+		let samples = self.0.eval(());
+		let i = self.1.eval(());
+		generate_quantization_fn(samples, i, input)
+	}
+}
+
+impl<N: Node<(), Output = u32> + Copy, M: Node<(), Output = u32> + Copy> Node<Image> for &GenerateQuantizationNode<N, M> {
+	type Output = Image;
+
+	fn eval(self, input: Image) -> Self::Output {
+		let samples = self.0.eval(());
+		let i = self.1.eval(());
+		generate_quantization_fn(samples, i, input)
+	}
+}
+
+impl<N: Node<(), Output = u32>, M: Node<(), Output = u32>> GenerateQuantizationNode<N, M> {
+	pub fn new(node: N, index: M) -> Self {
+		Self(node, index)
+	}
+}
+
+fn generate_quantization_fn(samples: u32, function: u32, input: Image) -> Image {
+	let max_energy = 16380.;
+	let data: Vec<f64> = input
+		.data
+		.iter()
+		.map(|x| vec![x.r() as f64, x.g() as f64, x.b() as f64])
+		.reduce(|mut acc, x| {
+			acc.extend_from_slice(&x);
+			acc
+		})
```

Comment:
Wouldn't a flat_map and collect make more sense.

Reply:
good catch